airbyte-integrations/connectors/source-postgres/src/main/kotlin/io/airbyte/integrations/source/postgres/PostgresSourceJdbcPartitionFactory.kt

```
@@ -134,97 +156,163 @@ open class PostgresSourceJdbcPartitionFactory(
+                            // snapshot ongoing
+                            PostgresSourceJdbcSplittableSnapshotWithXminPartition(
+                                selectQueryGenerator,
+                                streamState,
+                                Jsons.textNode(streamState.maybeCtid.toString()),
+                                null,
+                                sv.xmin,
+                                filenode,
+                                true
+                            )
+                        } else if (sv.xmin == streamState.cursorUpperBound) {
+                            // Incremental done
+                            null
+                        } else {
+                            filenode?.let { // Incremental ongoing
+                                PostgresSourceJdbcXminIncrementalPartition(
+                                    selectQueryGenerator,
+                                    streamState,
+                                    xminLowerBound = sv.xmin,
+                                    isLowerBoundIncluded = true,
+                                    xminUpperBound = streamState.cursorUpperBound,
+                                )
+                            }
+                                ?: error(
+                                    "Unexpected incremetal sync for a table ${stream.id} with no filenode."
+                                )
+                        }
+                    }
                 }
-                    ?: PostgresSourceJdbcUnsplittableCursorIncrementalPartition(
-                        selectQueryGenerator,
-                        streamState,
-                        cursor,
-                        cursorLowerBound = cursorCheckpoint,
-                        isLowerBoundIncluded = true,
-                        explicitCursorUpperBound = streamState.cursorUpperBound,
-                    )
             }
+            else -> TODO("Not implemented yet")
```

Is this only the CDC cases?

---

We just discussed that CDC won't use this. Maybe this should throw a different exception?